File: src/lib/whisper-message.ts
/**
 * Build a structured whisper message for warm transfer.
 * Per D-08: "[Name] calling about [job type]. [Emergency/Routine]. [1-line summary]."
 * Ported from src/lib/whisper-message.js — identical logic.
 */
export function buildWhisperMessage({
  callerName,
  jobType,
  urgency,
  summary,
}: {
  callerName?: string;
  jobType?: string;
  urgency?: string;
  summary?: string;
} = {}): string {
  const name = callerName || 'Unknown caller';
  const job = jobType || 'unspecified job';
  const tier = urgency === 'emergency' ? 'Emergency' : 'Routine';
  return `${name} calling about ${job}. ${tier}. ${summary || ''}`.trim();
}

File: src/lib/whisper_message.py
def build_whisper_message(
    caller_name: str | None = None,
    job_type: str | None = None,
    urgency: str | None = None,
    summary: str | None = None,
) -> str:
    name = caller_name or "Unknown caller"
    job = job_type or "unspecified job"
    tier = "Emergency" if urgency == "emergency" else "Routine"
    return f"{name} calling about {job}. {tier}. {summary or ''}".strip()
